src/simulation/m_monopole.fpp

```diff
@@ -69,19 +69,39 @@ contains
         @:ALLOCATE_GLOBAL(mag(1:num_mono), support(1:num_mono), length(1:num_mono), npulse(1:num_mono), pulse(1:num_mono), dir(1:num_mono), delay(1:num_mono), loc_mono(1:3, 1:num_mono), foc_length(1:num_mono), aperture(1:num_mono), support_width(1:num_mono))
 
         do i = 1, num_mono
+            do j = 1, 3
+                loc_mono(j, i) = mono(i)%loc(j)
+            end do
             mag(i) = mono(i)%mag
             support(i) = mono(i)%support
             length(i) = mono(i)%length
-            npulse(i) = mono(i)%npulse
-            pulse(i) = mono(i)%pulse
-            dir(i) = mono(i)%dir
-            delay(i) = mono(i)%delay
             foc_length(i) = mono(i)%foc_length
             aperture(i) = mono(i)%aperture
-            support_width(i) = mono(i)%support_width
-            do j = 1, 3
-                loc_mono(j, i) = mono(i)%loc(j)
-            end do
+            if (mono(i)%npulse == dflt_int) then
+                npulse(i) = 1
+            else
+                npulse(i) = mono(i)%npulse
+            end if
+            if (mono(i)%pulse == dflt_int) then
+                pulse(i) = 1
+            else
+                pulse(i) = mono(i)%pulse
+            end if
+            if (f_is_default(mono(i)%dir)) then
+                dir(i) = 1d0
+            else
+                dir(i) = mono(i)%dir
+            end if
+            if (f_is_default(mono(i)%delay)) then
+                delay(i) = 0d0
+            else
+                delay(i) = mono(i)%delay
+            end if
+            if (f_is_default(mono(i)%delay)) then
```

Comment:
typo here, should be `if (f_is_default(mono(i)%support_width)) then` I presume

Reply:
Sorry for the typo! Just fixed it